tests/test_visitInfo.py

```diff
@@ -360,6 +360,11 @@ def testPersistenceVersions(self):
         self._testFitsRead(self.data1, os.path.join(dataDir, "visitInfo-version-3.fits"), 3)
         self._testFitsRead(self.data1, os.path.join(dataDir, "visitInfo-version-4.fits"), 4)
 
+        # Check that reading a newer file raises a useful exception.
+        with self.assertRaisesRegex(TypeError,
+                                    r"Cannot read VisitInfo FITS version .* found version 999999"):
```

Comment:
```suggestion
                                    r"Cannot read VisitInfo FITS version > .* found version 999999"):
```

Reply:
It'd be good to include all parts of the string that don't vary in here, and leave only the variable version number for regex. I am not sure where the comma should go before the "found version", but I'll leave it to you whether to include it or not.